crates/rtmp/src/command_messages/netstream/reader.rs

```diff
@@ -0,0 +1,160 @@
+use std::convert::Infallible;
+use std::str::FromStr;
+
+use scuffle_amf0::{Amf0Decoder, Amf0Marker, Amf0Value};
+
+use super::NetStreamCommand;
+use super::define::NetStreamCommandPublishPublishingType;
+use crate::command_messages::error::CommandError;
+
+impl<'a> NetStreamCommand<'a> {
+    pub fn read(command_name: &str, decoder: &mut Amf0Decoder<'a>) -> Result<Option<Self>, CommandError> {
+        match command_name {
+            "play" => Ok(Some(Self::Play)),
+            "play2" => Ok(Some(Self::Play2)),
+            "deleteStream" => {
+                // skip command object
+                decoder.decode_with_type(Amf0Marker::Null)?;
+
+                let Amf0Value::Number(stream_id) = decoder.decode_with_type(Amf0Marker::Number)? else {
+                    unreachable!();
+                };
+
+                Ok(Some(Self::DeleteStream { stream_id }))
+            }
+            "closeStream" => Ok(Some(Self::CloseStream)),
+            "receiveAudio" => Ok(Some(Self::ReceiveAudio)),
+            "receiveVideo" => Ok(Some(Self::ReceiveVideo)),
+            "publish" => {
+                // skip command object
+                decoder.decode_with_type(Amf0Marker::Null)?;
+
+                let Amf0Value::String(publishing_name) = decoder.decode_with_type(Amf0Marker::String)? else {
+                    unreachable!();
+                };
+                let Amf0Value::String(publishing_type) = decoder.decode_with_type(Amf0Marker::String)? else {
+                    unreachable!();
+                };
+                let publishing_type = NetStreamCommandPublishPublishingType::from_str(&publishing_type).unwrap();
```

Comment:
For now we can just do .expect("error is infalliable")

when https://doc.rust-lang.org/std/result/enum.Result.html#method.into_ok is stable we can switch.